### DISCUSSION TITLE
Use different phones/cameras in one stream

### DISCUSSION THREAD

**Discussion by altunbayrakergun** [pedroSG94/RootEncoder#1463] [General]
I have a project in mind to make a single live broadcast across multiple or cameras. I need your opinions on this matter, both technically and in terms of feasibility. I aim to switch to the camera I want during the broadcast or combine multiple cameras at the same time. I am curious about RootEncoder's capabilities on this subject and need your opinions. Thank you.

**Comment by pedroSG94** [comment]
Hello,

I not sure if I understand your idea.
Are you talking about reproduce IP cameras? in this case. No, this project is for push/send video/audio data to a media server, not for reproduce IP cameras.

**Reply by altunbayrakergun** [reply]
Hi pedro, thank you for your quick reply. I was just wondering if I could create multiple surfaces, assign these surfaces to the connected cameras and switch between surfaces during the live broadcast when I need it. Maybe it's ridiculous, or this api might not have any talent for it. It was just an idea :)

**Comment by pedroSG94** [comment]
Hello,

What is a connected camera? an IP camera or a normal camera of the android device?

**Reply by altunbayrakergun** [reply]
Hi Pedro, 
If possible, I would like to use the camera on another android device. If not possible, a USB or Wi-Fi camera can also acceptable.

**Reply by pedroSG94** [reply]
Again, I haven't a response for my question. If your camera is an IP camera and you only want reproduce the video from that camera. you need find other library because this library is only for push streaming to a media server.
If you want push stream to a media server. You can add any VideoSource but you will need create your own source because the USB or WIFI cameras are not supported by default.

You have a branch with a USB VideoSource example created here (I can't give you support because it is only a demo that depend of other library so any bug related with that library should be asked in that library issues):
https://github.com/pedroSG94/RootEncoder/tree/feature/extra-video-source
You can compile this example, open rotation example, press menu button and then usb source. That should open the usb camera connected to the android device.

**Reply by altunbayrakergun** [reply]
Thank you for reply Pedro. As I said before. I'm not sure that should I use an ip camera or USB. And also thank you for example source. I'll try an ip camera using this source. So the next question is can I switch between an ip camera and device camera while streaming?

**Comment by pedroSG94** [comment]
Hello,

> So the next question is can I switch between an ip camera and device camera while streaming?

Yes, in the example code you can do exactly that using USB camera and device camera but you need use the same resolution to do that. IP camera is not available by default so you need find a way to create your custom VideoSource to support it.

**Reply by altunbayrakergun** [reply]
Thank you for your support Pedro. I'll try and if I can do it I'll let you know.
